hack/tools/validate_kindest_node.sh

```diff
@@ -4,8 +4,24 @@
 # Extract the version of kind, by removing the "${GOBIN}/kind-" prefix
 KIND=${KIND#${GOBIN}/kind-}
 
-# Get the version of the image
-KIND_VER=$(curl -L -s https://github.com/kubernetes-sigs/kind/raw/refs/tags/${KIND}/pkg/apis/config/defaults/image.go | grep -Eo 'v[0-9]+\.[0-9]+')
+GOMODCACHE=$(go env GOMODCACHE)
+
+REGEX='v[0-9]+\.[0-9]+'
+
+# Get the version of the image from the local kind build
+if [ -d "${GOMODCACHE}" ]; then
+    KIND_VER=$(grep -Eo "${REGEX}" ${GOMODCACHE}/sigs.k8s.io/kind@${KIND}/pkg/apis/config/defaults/image.go)
+fi
+
+# Get the version of the image from github
+if [ -z "${KIND_VER}" ]; then
+    KIND_VER=$(curl -L -s https://github.com/kubernetes-sigs/kind/raw/refs/tags/${KIND}/pkg/apis/config/defaults/image.go | grep -Eo "${REGEX}")
+fi
```

Comment:
is there a situation when we cannot resolve `KIND_VER` from the local cache? Can we remove this actually?